service/storage_service.cc

```diff
@@ -6175,6 +6175,151 @@ static bool increases_replicas_per_rack(const locator::topology& topology, const
+                        locator::global_tablet_id{table, tid}, req_id));
+            }
+            auto last_token = tmap.get_last_token(tid);
+            updates.emplace_back(
+                replica::tablet_mutation_builder(guard.write_timestamp(), table)
+                    .set_repair_task_info(last_token, repair_task_info)
+                    .build());
+        }
+
+        sstring reason = format("Repair tablet by API request tokens={} tablet_task_id={}", tokens, repair_task_info.tablet_task_id);
+        if (co_await exec_tablet_update(std::move(guard), std::move(updates), std::move(reason))) {
+            break;
+        }
+    }
+
+    co_await _topology_state_machine.event.wait([&] {
+        auto& tmap = get_token_metadata().tablets().get_tablet_map(table);
+        return std::all_of(tokens.begin(), tokens.end(), [&] (const dht::token& token) {
+            auto id = tmap.get_tablet_id(token);
+            return tmap.get_tablet_info(id).repair_task_info.tablet_task_id != repair_task_info.tablet_task_id;
+        });
+    });
+
+    auto duration = std::chrono::duration<float>(std::chrono::steady_clock::now() - start);
+    slogger.info("Finished tablet repair by API request table_id={} tokens={} all_tokens={} tablet_task_id={} duration={}",
+            table, tokens, all_tokens, repair_task_info.tablet_task_id, duration);
+
+    co_return res;
+}
+
+
+// Delete a tablet repair request by the given tablet_task_id
+future<> storage_service::del_repair_tablet_request(table_id table, locator::tablet_task_id tablet_task_id) {
+    auto holder = _async_gate.hold();
+
+    if (this_shard_id() != 0) {
+        // group0 is only set on shard 0.
+        co_return co_await container().invoke_on(0, [&] (auto& ss) {
+            return ss.del_repair_tablet_request(table, tablet_task_id);
+        });
+    }
+
+    if (!_feature_service.tablet_repair_scheduler) {
+        throw std::runtime_error("The TABLET_REPAIR_SCHEDULER feature is not enabled on the cluster yet");
+    }
+
+    slogger.info("Deleting tablet repair request by API request table_id={} tablet_task_id={}", table, tablet_task_id);
+    while (true) {
+        auto guard = co_await get_guard_for_tablet_update();
+
+        auto& tmap = get_token_metadata().tablets().get_tablet_map(table);
+        std::vector<canonical_mutation> updates;
+
+        co_await tmap.for_each_tablet([&] (locator::tablet_id tid, const locator::tablet_info& info) -> future<> {
+            auto& tinfo = tmap.get_tablet_info(tid);
+            auto& req_id = tinfo.repair_task_info.tablet_task_id;
+            if (req_id != tablet_task_id) {
+                co_return;
+            }
+            auto last_token = tmap.get_last_token(tid);
+            updates.emplace_back(
+                replica::tablet_mutation_builder(guard.write_timestamp(), table)
+                    .del_repair_task_info(last_token)
+                    .build());
```

Comment:
As I mentioned previously, it is possible we are not even in tablet_migration track. Is it safe to del_session outside the tablet_migration track? If it is safe, this indeed helps to speed up the abort of the existing tablet repair task, because since session is gone, we could use this to abort the existing task.

Reply:
It's not safe to delete session unconditionally, e.g. under the foot of regular migration. But I suggested to check for it: `if transition == repair`. If transition is repair, session will be deleted for that repair track. Updates to group0 are linearized.

Reply:
Done  in latest version.